packages/webdriverio/tests/middleware.test.js

```diff
@@ -0,0 +1,61 @@
+import logger from '@wdio/logger'
+import { remote } from '../src'
+import request from 'request'
+
+const { warn  } = logger()
+const waitForExist = require('../src/commands/element/waitForExist')
```

Comment:
I think we can use `jest` mocking capabilities here:
```js
jest.mock('../src/commands/element/waitForExist', () => .mockImplementation ( ( ) => { throw new Error(`Promise was rejected with the following reason`)})
```

Reply:
Done.